src/conn/conn_compact.c

```
@@ -721,6 +721,10 @@ __wti_background_compact_server_create(WT_SESSION_IMPL *session)
     if (F_ISSET(conn, WT_CONN_IN_MEMORY | WT_CONN_READONLY))
         return (0);
 
+    /* Disable background compact server in disagg mode. */
+    if (__wt_conn_is_disagg(session))
```

Can we return an error here explicitly to the user if the user configured to enable it in disagg?

---

Can we also create a python test to ensure the error is thrown to the caller?

---

This is inside the background compaction server thread creation, not in a user side call, so just skip is `OK` here. For `session_compact` there already have a `ENOTSUP` return, I'll add a test case for that.

---

I mean if user specify background_compact=(enabled=true) in wt open, we should throw an error.

---

After check, seems there's not a config like this to control the enable of `background_compact`.

---

See [here](https://source.wiredtiger.com/develop/struct_w_t___s_e_s_s_i_o_n.html#aafa7a12a4891a5bfdc98673a5b8f9c69) for examples on how to interact with background compaction through the `session::compact` API.
You can also check the `compact_util.py` file to check the different testing helpers.
I am expecting a test to be similar to this:
```
        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
            lambda: self.session.compact(None, 'background=true'),
            '/Operation not supported/')
```
As long as we report `ENOTSUP` in the relevant flow.

---

As [in this PR change](https://github.com/wiredtiger/wiredtiger/blob/710c1b9ed40367be47e319d3bb31738135e084fa/src/session/session_compact.c#L408), the code will immediately fail without further explain the config string, so there's not a execution route different with various configs.

---

ok, gotcha